tests/restart-song.spec.ts

```diff
@@ -11,31 +10,23 @@ test.beforeEach(async ({ page, context, browser }) => {
   await mockSongs({ page, context });
 });
 
+const songID = 'e2e-multitrack-polish-1994';
+
 test('should restart the song and the scores', async ({ page }) => {
   await page.goto('/?e2e-test');
   await pages.landingPage.enterTheGame();
-  await page.getByTestId('advanced').click();
-  await page.getByTestId('save-button').click();
-
-  await page.getByTestId('sing-a-song').click();
-  await expect(page.getByTestId('lang-Polish')).toBeVisible();
-  await page.getByTestId('close-exclude-languages').click();
-
-  await expect(page.getByTestId('song-e2e-single-english-1995')).toBeVisible();
-  await navigateWithKeyboard(page, 'song-e2e-multitrack-polish-1994');
+  await pages.inputSelectionPage.selectAdvancedSetup();
+  await pages.advancedConnectionPage.saveAndGoToSing();
+  await pages.mainMenuPage.goToSingSong();
+  await pages.songLanguagesPage.continueAndGoToSongList();
+  await expect(pages.songListPage.getSongElement(songID)).toBeVisible();
+  await pages.songListPage.navigateToSongWithKeyboard(songID);
   await page.keyboard.press('Enter');
-
-  await page.getByTestId('next-step-button').click();
-  await page.getByTestId('play-song-button').click();
-
-  await expect(async () => {
-    const p1score = await page.getByTestId('players-score').getAttribute('data-score');
-
-    expect(parseInt(p1score!, 10)).toBeGreaterThan(100);
-  }).toPass({ timeout: 15_000 });
+  await pages.songPreviewPage.goNext();
+  await pages.songPreviewPage.playTheSong();
+  await pages.gamePage.waitForPlayersScoreToBeGreaterThan();
 
   await page.keyboard.press('Backspace');
-  await page.getByTestId('button-restart-song').click();
-
-  await expect(page.getByTestId('players-score')).toHaveAttribute('data-score', '0', { timeout: 15_000 });
+  await pages.gamePage.restartSong();
+  await expect(pages.gamePage.playersScoreElement).toHaveAttribute('data-score', '0', { timeout: 15_000 });
```

Comment:
could we have a new method in the game page PO `epectPlayersScoreToBe` to hide this "to have attribute" detail from the test?

Reply:
Sure, done. Thanks :)